ghauth.js

```diff
@@ -35,123 +34,227 @@ function newlineify (len, str) {
   return s
 }
 
-async function createAuth (options) {
-  const reqOptions = {
+function sleep (s) {
+  const ms = s * 1000
+  return new Promise(resolve => setTimeout(resolve, ms))
+}
+
+// prompt the user for credentials
+async function prompt (options) {
+  const promptName = options.promptName || defaultPromptName
+  const scopes = options.scopes || defaultScopes
+  const passwordReplaceChar = options.passwordReplaceChar || defaultPasswordReplaceChar
+  const githubHost = options.githubHost || defaultGithubHost
+  const isEnterprise = githubHost !== defaultGithubHost
+  const deviceCodeUrl = `https://${githubHost}/login/device/code`
+  const fallbackDeviceAuthUrl = `https://${githubHost}/login/device`
+  const accessTokenUrl = `https://${githubHost}/login/oauth/access_token`
+  const oauthAppsBaseUrl = `https://${githubHost}/settings/connections/applications`
+  const apiUrl = isEnterprise ? `https://api.${githubHost}` : `https://${githubHost}/api/v3`
```

Comment:
Do you think this will do it for GH enterprise?

No idea about the rest of the URLS wrt GH Enterprise.  I reached out to GitHub Support to find out if it even supports the new device flow. 

I guess worse case scenario, enterprise users can use the PAT flow. Still, ghauth now needs to hit the user endpoint.